### ISSUE TITLE
Does support use tpu to pretrain longformer?

### ISSUE THREAD

**Issue by RyanHuangNLP** [allenai/longformer#101]
Now supports using tpu to pre-train longformer？

**Comment by ibeltagy** [comment]
HF recently added a Tensorflow version, so it might work on TPUs but we haven't tried. 
We also have a WIP version that runs on pytroch-xla. I can clean it up and push it if you want.

**Comment by RyanHuangNLP** [comment]
@ibeltagy Yes, I want the WIP version that runs on pytroch-xla, and I will take a try with Tensorflow version

**Comment by RyanHuangNLP** [comment]
@ibeltagy I also wonder does eight V100 32G is enough to train a longformer large?

**Comment by ibeltagy** [comment]
> pytroch-xla
We have a working version in this branch https://github.com/allenai/longformer/tree/trainer. We will clean it up and release it soon, but it is good as is.

> V100 32G
Yes

**Comment by mabdullah1994** [comment]
Hi. I'm trying to fine-tune LongformerForSequenceClassification on PyTorch XLA. The environment is GCP TPU and I'm getting the following error:

IndexError: tuple index out of range

Please advice. Thanks!

**Comment by ibeltagy** [comment]
We have WIP here https://github.com/allenai/longformer/tree/trainer. It is not polished and we don't have immediate plans to release it.

**Comment by mabdullah1994** [comment]
Ok thanks. Will look into it! @ibeltagy
